### PR TITLE
Extract Test Code to child modules

### PR DESCRIPTION
**Pull request by darthscsi**
Extract simulation constructs to child modules which will be instantiated with a bind.

[TODO] Rebase without the bind op PR.
[TODO] Generate a bind statement rather than using an attribute on an instance.

### CODE REVIEW

File: include/circt/Dialect/RTL/RTLStructure.td [hunk 1]
```diff
@@ -19,7 +19,8 @@ def RTLModuleOp : RTLOp<"module",
     name, a list of ports, and a body that represents the connections within
     the module.
   }];
-  let arguments = (ins StrArrayAttr:$argNames, StrArrayAttr:$resultNames);
```
Comment (lattner) [inline include/circt/Dialect/RTL/RTLStructure.td:22]: Please split stuff like this out to its own patch since it is logically separate.

That said, I don't think we need this - the reason that extmodule has this is because you can have multiple extmodule instances with the same verilog name (due to parameters).  RTLModuleOp doesn't have that going on.

File: include/circt/Dialect/SV/SVStatements.td [hunk 2]
```diff
@@ -474,3 +481,27 @@ def CoverOp : SVOp<"cover", [ProceduralOp]> {
   let results = (outs);
   let assemblyFormat = "attr-dict $property `:` type($property)";
 }
+
+def BindOp : SVOp<"bind", [IsolatedFromAbove, HasParent<"mlir::ModuleOp">]> {
```
Comment (lattner) [inline include/circt/Dialect/SV/SVStatements.td:485]: Oh, maybe this just has other stuff mixed in?  BindOp should (is?) definitely a separable thing.

Comment (seldridge) [reply include/circt/Dialect/SV/SVStatements.td:485]: Yeah, this patch set is stacked onto https://github.com/llvm/circt/pull/982. The `BindOp` stuff should land first. GitHub annoyingly has no good support for indicating this. :(

Comment (lattner) [reply include/circt/Dialect/SV/SVStatements.td:485]: Aha, got it.  Let's get it in! :-)
